storage/integration_test.go

```diff
@@ -1069,13 +1069,42 @@ func testObjectIterator(t *testing.T, bkt *BucketHandle, objects []string) {
 	if !ok {
 		t.Errorf("ObjectIterator.Next: %s", msg)
 	}
+	m := make(map[string][]*ObjectAttrs)
+	for i, name := range names {
+		// StartOffset takes the value of object names, the result must be for:
+		// ― obj/with/slashes: obj/with/slashes, obj1, obj2
+		// ― obj1: obj1, obj2
+		// ― obj2: obj2.
+		m[name] = attrs[i:]
+		msg, ok := itesting.TestIterator(m[name],
+			func() interface{} { return bkt.Objects(ctx, &Query{StartOffset: name}) },
+			func(it interface{}) (interface{}, error) { return it.(*ObjectIterator).Next() })
+		if !ok {
+			t.Errorf("ObjectIterator.Next: %s", msg)
+		}
+		// EndOffset takes the value of object names, the result must be for:
+		// ― obj/with/slashes: ""
+		// ― obj1: obj/with/slashes
+		// ― obj2: obj/with/slashes, obj1.
+		m[name] = attrs[:i]
+		msg, ok = itesting.TestIterator(m[name],
+			func() interface{} { return bkt.Objects(ctx, &Query{EndOffset: name}) },
+			func(it interface{}) (interface{}, error) { return it.(*ObjectIterator).Next() })
+		if !ok {
+			t.Errorf("ObjectIterator.Next: %s", msg)
+		}
+	}
 	// TODO(jba): test query.Delimiter != ""
 }
 
 func testObjectsIterateSelectedAttrs(t *testing.T, bkt *BucketHandle, objects []string) {
 	// Create a query that will only select the "Name" attr of objects, and
-	// invoke object listing.
-	query := &Query{Prefix: ""}
+	// invoke object listing, with given offsets. The last object will be excluded.
```

Comment:
Let's not have this test include StartOffset/EndOffset just to have a variety of queries (you can leave it in the test below though).

Reply:
Okay, I left it in the testObjectsIterateAllSelectedAttrs